velox/common/caching/tests/SsdFileTest.cpp

```diff
@@ -795,6 +798,135 @@ TEST_F(SsdFileTest, ssdReadWithoutChecksumCheck) {
 #endif
 }
 
+TEST_F(SsdFileTest, faultyFiles) {
+  constexpr int64_t kSsdSize = 16 * SsdFile::kRegionSize;
+  const uint64_t checkpointIntervalBytes = 5 * SsdFile::kRegionSize;
+  const auto retainFile = StringIdLease(fileIds(), "faultyFiles.Retained");
+  const auto evictFile = StringIdLease(fileIds(), "faultyFiles.Evicted");
+
+  FLAGS_ssd_odirect =
+      false; // tmpfs does not support O_DIRECT, so turn this off for testing.
+  cache_ = AsyncDataCache::create(memory::memoryManager()->allocator());
```

Comment:
You might want to put this into a utility say setupCache(bool withFaultyFs = false)